### PR TITLE
LPS-133868 Add allowedFileExtensions property

### PR DESCRIPTION
**Pull request by MateusCastro**

### CODE REVIEW

**Comment by MateusCastro** [comment]
ci:test:sf

**Comment by MateusCastro** [comment]
ci:test:relevant

File: modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/css/main.scss [hunk 1]
```diff
@@ -155,4 +155,18 @@
 			max-height: 100%;
 		}
 	}
+
+	.file-entries-list .has-error {
```
Comment (kevenleone) [inline modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/css/main.scss:159]: This class should be placed before `.envelope-list`

To respect the alphabetical order

File: modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/css/main.scss [hunk 2]
```diff
@@ -155,4 +155,18 @@
...
+		background-color: #feefef;
+		border: 1px solid #f48989;
+
+		.quick-action-menu {
+			background-color: #feefef;
+		}
+
+		.text-error {
```
Comment (kevenleone) [inline modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/css/main.scss:167]: This class should be placed before `.quick-action-menu`

To respect the alphabetical order

**Review by kevenleone — APPROVED** [review]
Unless we need to change anything else, everything is good to me, my review comments could be changed later

File: modules/apps/digital-signature/digital-signature-lang/src/main/resources/content/Language.properties [hunk 3]
```diff
@@ -1,11 +1,14 @@
 account's-base-uri=Account's Base URI
+add-document=Add Document
 add-recipient=Add Recipient
+added-documents=Added Documents
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-lang/src/main/resources/content/Language.properties:4]: Replace by **Documents Added**

File: modules/apps/digital-signature/digital-signature-lang/src/main/resources/content/Language.properties [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
 api-account-id=API Account ID
 api-username=API User ID
 category.digital-signature=Digital Signature
 collect-digital-signature=Collect Digital Signature
 digital-signature=Digital Signature
 digital-signature-configuration-name=Digital Signature Configuration
+document-already-exist=Document already exist.
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-lang/src/main/resources/content/Language.properties:11]: Replace by **Document Already Exists** (Title)

File: modules/apps/digital-signature/digital-signature-lang/src/main/resources/content/Language.properties [hunk 5]
```diff
@@ -21,6 +24,7 @@ recipient-full-name=Recipient Full Name
 rsa-private-key=RSA Private Key
 the-document-does-not-have-a-preview=The document does not have a preview.
 this-field-exceeded-x-characters=This field exceeded {0} characters.
+this-file-extension-is-not-supported-by-docusign=This file extension is not supported by DocuSign
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-lang/src/main/resources/content/Language.properties:27]: This is a sentence, insert a period at the end.
**This file extension is not supported by DocuSign.**

File: modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/collect_digital_signature/view.jsp [hunk 6]
```diff
@@ -31,6 +31,8 @@ if (digitalSignatureTitle != null) {
 		module="js/pages/CollectDigitalSignature"
 		props='<%=
 			HashMapBuilder.<String, Object>put(
+				"allowedFileExtension", StringUtil.merge(DigitalSignatureConstants.ALLOWED_FILE_EXTENSIONS)
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/collect_digital_signature/view.jsp:34]: Follow the same name of constant (plural): 
**"allowedFileExtensions", StringUtil.merge(DigitalSignatureConstants.ALLOWED_FILE_EXTENSIONS**

File: modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/digital-signature-form/DigitalSignatureForm.js [hunk 7]
```diff
@@ -35,11 +41,15 @@ const defaultRecipient = {
...
 
 const DigitalSignatureForm = ({
-	fileEntryIds,
+	fileEntries = [],
 	history,
 	showDocumentLibraryInput,
 }) => {
-	const {baseResourceURL, portletNamespace} = useContext(AppContext);
+	const {
+		allowedFileExtension,
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/digital-signature-form/DigitalSignatureForm.js:49]: **allowedFileExtensions**

File: modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/digital-signature-form/DigitalSignatureForm.js [hunk 8]
```diff
@@ -35,11 +41,15 @@ const defaultRecipient = {
 };
 
 const DigitalSignatureForm = ({
-	fileEntryIds,
+	fileEntries = [],
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/digital-signature-form/DigitalSignatureForm.js:44]: Is this array of IDs?

File: modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/digital-signature-form/DigitalSignatureForm.js (hunk 7 above)
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/digital-signature-form/DigitalSignatureForm.js:49]: **allowedFileExtension**

File: modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/form/FormValidation.js [hunk 9]
```diff
@@ -50,4 +62,4 @@ const validate = (fields, values) => {
 	return errors;
 };
 
-export {isEmail, maxLength, required, validate};
+export {isEmail, maxLength, required, validate, withInvalidExtension};
```
Comment (joseabelenda) [inline modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/form/FormValidation.js:65]: The same of above

**Review by joseabelenda — CHANGES_REQUESTED** [review]
Please, see my comments and suggestions.
Thanks!

File: modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/digital-signature-form/DigitalSignatureForm.js (hunk 8 above)
Comment (kevenleone) [reply modules/apps/digital-signature/digital-signature-web/src/main/resources/META-INF/resources/js/components/digital-signature-form/DigitalSignatureForm.js:44]: Nop, Array of Object

**Comment by kevenleone** [comment]
Just started reviewing :)

File: modules/apps/digital-signature/digital-signature-lang/src/main/resources/content/Language.properties (hunk 4 above)
Comment (MateusCastro) [reply modules/apps/digital-signature/digital-signature-lang/src/main/resources/content/Language.properties:11]: Hi Abelenda, this is an error message and not a title.

**Comment by MateusCastro** [comment]
ci:test:sf

**Comment by MateusCastro** [comment]
ci:test:relevant

**Comment by kevenleone** [comment]
Just started reviewing :)

**Review by joseabelenda — APPROVED** [review]
LGTM.
Thanks @MateusCastro

**Comment by kevenleone** [comment]
ci:forward

**Pull request closed without merging**